src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinder.cs

```diff
@@ -760,7 +760,7 @@ private Expr CreateArgumentEXPR(ArgumentObject argument, LocalVariableSymbol loc
                 flags, name, typeArgumentsAsTypeArray, kind, callingType, null, null, new CMemberLookupResults(
                     _semanticChecker.getBSymmgr().AllocParams(callingTypes.Count, callingTypes.ToArray()),
                     name));
-            if (callingObject.isCLASS())
+            if (callingObject is ExprClass)
```

Comment:
need to make sure that the types that are used for "is" compare are sealed. That would make type check much cheaper. Otherwise checking the "Kind" might be preferable.